File: tests/machine/x/clj/string_prefix_slice.clj
(ns main)

(defn _equal [a b]
  (cond
    (and (sequential? a) (sequential? b))
      (and (= (count a) (count b)) (every? true? (map _equal a b)))
    (and (map? a) (map? b))
      (and (= (count a) (count b))
           (every? (fn [k] (_equal (get a k) (get b k))) (keys a)))
    (and (number? a) (number? b))
      (= (double a) (double b))
    :else
      (= a b)))

(declare prefix s1 s2)

(defn -main []
  (def prefix "fore") ;; string
  (def s1 "forest") ;; string
  (println (_equal (subs s1 0 (count prefix)) prefix))
  (def s2 "desert") ;; string
  (println (_equal (subs s2 0 (count prefix)) prefix))
)

(-main)

File: tests/machine/x/fs/string_prefix_slice.fs
open System

exception Break
exception Continue

let prefix: string = "fore"
let s1: string = "forest"
printfn "%s" s1.Substring(0, List.length prefix - 0) = prefix
let s2: string = "desert"
printfn "%s" s2.Substring(0, List.length prefix - 0) = prefix
